client/my-sites/sharing/controller.js

```diff
@@ -18,17 +18,25 @@ import { sectionify } from 'lib/route';
 import Sharing from './main';
 import SharingButtons from './buttons/buttons';
 import SharingConnections from './connections/connections';
-import sites from 'lib/sites-list';
-import utils from 'lib/site/utils';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { canCurrentUser, isJetpackModuleActive } from 'state/selectors';
+import { getSiteSettings } from 'state/site-settings/selectors';
+import { requestSiteSettings } from 'state/site-settings/actions';
+import { reduxDispatch } from 'lib/redux-bridge/index';
+import { isJetpackSite, getSiteSlug, getSiteOption } from 'state/sites/selectors';
+import versionCompare from 'lib/version-compare';
 
 const analyticsPageTitle = 'Sharing';
 
 export const layout = context => {
-	const site = sites().getSelectedSite();
 	const { contentComponent, path, store } = context;
+	const state = store.getState();
 
-	if ( site && ! site.settings && utils.userCan( 'manage_options', site ) ) {
-		site.fetchSettings();
+	const siteId = getSelectedSiteId( state );
+	const siteSettings = getSiteSettings( state, siteId );
+
+	if ( siteId && ! siteSettings && canCurrentUser( state, siteId, 'manage_options' ) ) {
+		reduxDispatch( requestSiteSettings( siteId ) );
```

Comment:
You shouldn't need to use the `redux-bridge` in a `page` controller like this: you can directly call `store.dispatch( ... )`.

The bridge is only for modules that can't access the store in any other way, i.e., Flux stores.

Reply:
Ah, didn't know. Thanks!